### PR TITLE
Only collect archives in the case of aborted or failed runs.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Our archives are currently quite large (though we are working to reduce their size).  We can save a lot of machine space if we only collect archives for when things actually go wrong.  No one needs them for when things go right :)

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-infrastructure

**Comment by CyrusNajmabadi** [comment]
tagging @mmitche

File: netci.groovy [hunk 1]
```diff
@@ -58,13 +58,20 @@ static void addWrappers(def myJob) {
 static void addArtifactArchiving(def myJob, String patternString, String excludeString) {
   myJob.with {
     publishers {
-      archiveArtifacts {
```
Comment (davkean) [inline netci.groovy]: How did we get this syntax, this is basically
```
publishers {

    flexiblePublish{
        publishers {

        }
   }
}
```

Comment (CyrusNajmabadi) [inline netci.groovy]: Yes.   I ran this by @mmitche  to verify that, and he said that was appropriate.

However, from looking at the jenkins DSL, it looks like "publishers" should actually nest under "conditionalAction" (though they can be siblings, it's just deprecated).

Comment (CyrusNajmabadi) [inline netci.groovy]: I've since updated things based on the DSL here: https://jenkinsci.github.io/job-dsl-plugin/#path/job-publishers-flexiblePublish

**Comment by mmitche** [comment]
LGTM.  I'll merge it later this weekend.

**Comment by davkean** [comment]
:+1: Let's merge this in soon.

**Pull request merged**